### ISSUE TITLE
Don't re-render page when auth token refreshes

### ISSUE THREAD

**Issue by Powersource** [Edgeryders-Participio/realities#204]

**Comment by Powersource** [comment]
I have an idea for a possible simple fix :thinking:

**Comment by Powersource** [comment]
Related code

https://github.com/Edgeryders-Participio/realities/blob/1cf5cfad50b6fed85c325335a070c4df47ea631d/ui/src/App/App.js#L33-L52

The problem I think is that whenever it re-renders, it creates a new apolloClient on line 48. I think it does the rerendering when there's a new token because of the useOidc use. That's good to some degree because it re-establishes the websocket connection, otherwise it'd continue using the old token for the subscriptions. And it used to be necessary to create the new apolloClient because we used to pass in the token as an argument. Now we fixed so that it can dynamically get the token from sessionstorage when it's constructing the headers for a request.

So basically if we memoize apolloClient it probably won't do the rerendering (to note is I don't think it's techincally the rerendering that makes everything spin and be cleared, it's since the apolloClient's cache gets reset it needs to refetch everything and local cache state is reset) every 5 min but subscriptions will break after 5 min until we can find a better solution for that. Probably a worthy tradeoff.

**Comment by aerugo** [comment]
That’s an okay trade off for now, and opening another issue to remember
fixing the subscriptions.

**Comment by Powersource** [comment]
Will wait with opening an issue about the subscriptions because stuff seemed to continue working ok for a bit now (might be that you don't really need auth so far for anything the subscriptions are used for so they continue without a problem even if the token's old?), and we have a TODO comment about it so we won't completely forget.
